crates/uv-auth/src/lib.rs

```diff
@@ -35,3 +35,61 @@ pub fn store_credentials_from_url(url: &Url) -> bool {
         false
     }
 }
+
+/// Populate the global authentication store with credentials from the environment, if any.
+///
+/// Respects `UV_AUTH_URLS` with whitespace separated URLs.
+///
+/// Supports formats like:
+///
+/// - `UV_AUTH_URLS="username:password@hostname"`
+/// - `UV_AUTH_URLS="username@hostname"`
+/// - `UV_AUTH_URLS="username:password@hostname username:password@other-hostname"`
+/// - `UV_AUTH_URLS="username:password@hostname/path"`
+///
+/// Only `https` URLs are supported, URLs with other schemes are ignored.
+///
+/// Populates the URL prefix cache in the global [`CredentialsCache`]. The credentials are not
+/// added the the realm-level cache to ensure that a URL with a path is not used unless a request
+/// is made to a child of the provided URL.
+pub fn store_credentials_from_environment() -> bool {
+    let Some(urls) = std::env::var_os("UV_AUTH_URLS") else {
+        return false;
+    };
+
+    let mut populated = false;
+    for mut url in urls
+        .to_string_lossy()
+        .split_ascii_whitespace()
+        .filter_map(parse_url_from_env)
+    {
+        if let Some(credentials) = Credentials::from_url(&url) {
+            CREDENTIALS_CACHE.insert_url(&url, Arc::new(credentials));
+            // Redact the password for display
+            if url.password().is_some() {
+                let _ = url.set_password(Some("***"));
+            }
+            debug!("Added credentials for `{url}`");
+            populated = true;
+        } else {
+            debug!("Ignoring URL without credentials in `UV_AUTH_URLS`: {url}");
+        }
+    }
+
+    populated
+}
+
+/// Parse a URL, allowing the scheme to be missing (and inferred as `https://`)
+fn parse_url_from_env(url: &str) -> Option<Url> {
+    if url.starts_with("https://") {
+        Url::parse(url)
+    } else if url.starts_with("http://") {
+        debug!("Ignoring insecure URL in `UV_AUTH_URLS`: {url}");
```

Comment:
Should this be a `warn!`?

Reply:
Can't use our `warn_user` macros here, but we could use a warn tracing level — just doesn't mean much yet. Seems reasonable though.

Reply:
```suggestion
        warn!("Ignoring insecure URL in `UV_AUTH_URLS`: {url}");
```